File: functional test automation/webdriver/file upload/upload_file.py
# -*- coding: utf-8 -*-

# подключаем необходимые модули
from selenium import webdriver
import os

# создаём объект WebDriver для браузера chrome
driver = webdriver.Chrome()

# открываем страницу с формой загрузки файла
driver.get("file://" + os.getcwd() + "/index.html")

# находим элемент <input type="file">
element = driver.find_element_by_id("file")

# заполняем элемент путём до загружаемого файла
element.send_keys(os.getcwd() + "/lenna.png")

# находим элемент <input type="file">
element = driver.find_element_by_id("submit")

# нажимаем на элемент (отправляем форму)
element.click()

# закрываем браузер
driver.quit()

File: functional test automation/webdriver/file upload/upload_file.rb
# encoding: utf-8

# подключам необходимый gem
require 'selenium-webdriver'

# создаём объект WebDriver для браузера chrome
driver = Selenium::WebDriver.for :chrome

# открываем страницу с формой загрузки файла
driver.get "file://#{File.expand_path(File.dirname(__FILE__))}/index.html"

# находим элемент <input type="file">
element = driver.find_element :id, 'file'

# заполняем элемент путём до загружаемого файла
element.send_keys "#{File.expand_path(File.dirname(__FILE__))}/lenna.png"

# находим элемент <input type="file">
element = driver.find_element :id, 'submit'

# нажимаем на элемент (отправляем форму)
element.click

# закрываем браузер
driver.close
